app/test/max_tx_size_integration_test.go

```diff
@@ -0,0 +1,68 @@
+package app_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/celestiaorg/celestia-app/v4/app"
+	"github.com/celestiaorg/celestia-app/v4/app/encoding"
+	apperrors "github.com/celestiaorg/celestia-app/v4/app/errors"
+	appv4 "github.com/celestiaorg/celestia-app/v4/pkg/appconsts/v4"
+	"github.com/celestiaorg/celestia-app/v4/test/util/blobfactory"
+	"github.com/celestiaorg/celestia-app/v4/test/util/random"
+	"github.com/celestiaorg/celestia-app/v4/test/util/testnode"
+)
+
+// TestTxsOverMaxTxSizeGetRejected tests that transactions over the max tx size get rejected
+// by the application even if the validator node's local mempool config allows them.
+func TestTxsOverMaxTxSizeGetRejected(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping max tx size integration test in short mode.")
+	}
+
+	accounts := testnode.RandomAccounts(10)
+	// Set the max block size to 8MB
+	cparams := testnode.DefaultConsensusParams()
+	cparams.Block.MaxBytes = 8_388_608 // 8MiB
+	cfg := testnode.DefaultConfig().WithFundedAccounts(accounts...).WithConsensusParams(cparams)
+
+	// Set the max tx bytes to 3MiB in the node's mempool
+	mempoolMaxTxBytes := appv4.MaxTxSize + 1_048_576 // 2MiB + 1MiB
```

Comment:
[blocking]

apologies I don't think I was clear enough in the last review. I don't think we should be relying only on this config, if validators change this then they will allow txs in their mempool that could never make it in a block. so I think we still need the antehandler.

Reply:
In this test, I'm increasing the MaxTxBytes in node's mempool config to be larger than app's threshold and we see that txs get rejected with the expected log and error code.